tests/KeyValueStoreTests.cpp

```diff
@@ -0,0 +1,58 @@
+#include "../src/key_value_store/core/KeyValueStore.h"
+
+#include <mutex>
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+TEST(KeyValueStoreTest, basic_put_pass) {
```

Comment:
Document unit tests

Reply:
Sorry, not sure what this means. Can you please elaborate?

Reply:
You should add comments before each unit test describing what that test is testing

Reply:
My bad, added the comments!